official/vision/beta/projects/yolo/modeling/decoders/yolo_decoder_test.py

```diff
@@ -0,0 +1,148 @@
+
+from tensorflow.python.distribute import combinations
+from tensorflow.python.distribute import strategy_combinations
+from official.vision.beta.projects.yolo.modeling.decoders import yolo_decoder as decoders
+
+
+class YoloDecoderTest(parameterized.TestCase, tf.test.TestCase):
+
+  @parameterized.parameters('a', 'b', 'c')
+  def test_network_creation(self, version):
+    """Test creation of ResNet family models."""
+    tf.keras.backend.set_image_data_format('channels_last')
+    input_shape = {
+        '3': [1, 52, 52, 256],
+        '4': [1, 26, 26, 512],
+        '5': [1, 13, 13, 1024]
+    }
+    decoder = build_yolo_decoder(input_shape, version)
+
+    inputs = {}
+    for key in input_shape.keys():
+      inputs[key] = tf.ones(input_shape[key], dtype=tf.float32)
+
+    endpoints = decoder.call(inputs)
+
+    for key in endpoints.keys():
+      self.assertAllEqual(endpoints[key].shape.as_list(), input_shape[key])
+
+  @combinations.generate(
+      combinations.combine(
+          strategy=[
+              strategy_combinations.tpu_strategy,
+              strategy_combinations.one_device_strategy_gpu,
+          ],
+          use_sync_bn=[False, True],
+      ))
+  def test_sync_bn_multiple_devices(self, strategy, use_sync_bn):
+    """Test for sync bn on TPU and GPU devices."""
+
+    tf.keras.backend.set_image_data_format('channels_last')
+
+    with strategy.scope():
+      input_shape = {
+          '3': [1, 52, 52, 256],
+          '4': [1, 26, 26, 512],
+          '5': [1, 13, 13, 1024]
+      }
+      decoder = build_yolo_decoder(input_shape, 'c')
+
+      inputs = {}
+      for key in input_shape.keys():
+        inputs[key] = tf.ones(input_shape[key], dtype=tf.float32)
+
+      _ = decoder.call(inputs)
+
+  @parameterized.parameters(1, 3, 4)
+  def test_input_specs(self, input_dim):
+    """Test different input feature dimensions."""
+    tf.keras.backend.set_image_data_format('channels_last')
+
+    input_shape = {
+        '3': [1, 52, 52, 256],
+        '4': [1, 26, 26, 512],
+        '5': [1, 13, 13, 1024]
+    }
+    decoder = build_yolo_decoder(input_shape, 'c')
+
+    inputs = {}
+    for key in input_shape.keys():
+      inputs[key] = tf.ones(input_shape[key], dtype=tf.float32)
+    _ = decoder(inputs)
+
+  def test_serialize_deserialize(self):
+    """Create a network object that sets all of its config options."""
+    tf.keras.backend.set_image_data_format('channels_last')
+
+    input_shape = {
+        '3': [1, 52, 52, 256],
+        '4': [1, 26, 26, 512],
+        '5': [1, 13, 13, 1024]
+    }
+    decoder = build_yolo_decoder(input_shape, 'c')
+
+    inputs = {}
+    for key in input_shape.keys():
+      inputs[key] = tf.ones(input_shape[key], dtype=tf.float32)
+
+    _ = decoder(inputs)
+
+    a = decoder.get_config()
+
+    b = decoders.YoloDecoder.from_config(a)
+
+    print(a)
+    self.assertAllEqual(decoder.get_config(), b.get_config())
+
+
+def build_yolo_decoder(input_specs, type):
+  if type == 'a':
```

Comment:
Suggest you do the following:

- Extract the shared dictionary
- Update the dictionary with varying parameters using conditional logic
- Make model *after* the conditions
- Remove the `model` variable, just `return decoders.YoloDecoder(**params)`

Also:
- `a` and `b` are uninformative names
- consider a default value for `type` parameter, like `type='default'`

Reply:
These are just names for the test cases. They are renamed to the size of the decoder, such as 1, 6, and 6spp.